### ISSUE TITLE
Support Renovating Flux's HelmRelease CRD inside Helm Chart

### ISSUE THREAD

**Issue by cwrau** [renovatebot/renovate#18423]
### What would you like Renovate to be able to do?

To find and support Flux's HelmRelease CRD inside Helm Chart;

We have [a  Helm Chart](https://github.com/teutonet/teutonet-helm-charts/tree/main/charts/base-cluster) that template Flux HelmRelease CRDs (e.g. [ingress-nginx](https://github.com/teutonet/teutonet-helm-charts/blob/main/charts/base-cluster/templates/ingress/nginx.yaml))

### If you have any ideas on how this should be implemented, please tell us here.

Either renovate could look at the raw template files, which wouldn't necessarily work, as at least in our case the HelmRepositories are not static and therefore cannot be found this way. Also, the HelmReleases might be dynamically generated as well 😅

But renovate could template the HelmChart, probably with a `renovate-values.yaml`, and analyse the resulting files.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by viceice** [comment]
would that values file part of the repo, so renovate finds it side by side with the templates?

otherwise your best option is to use the regex manager.

**Comment by cwrau** [comment]
> would that values file part of the repo, so renovate finds it side by side with the templates?

Yes, it has to, as all charts need their own custom values to achieve this

**Comment by danports** [comment]
You use a Helm chart to deploy a Flux `HelmRelease` resource that deploys a Helm chart? I think you just broke my brain a little... 🤯 Why not just deploy the Helm chart directly from a `HelmRelease` stored in Git, a.k.a. the standard Flux workflow?

**Comment by cwrau** [comment]
> You use a Helm chart to deploy a Flux `HelmRelease` resource that deploys a Helm chart? I think you just broke my brain a little... 🤯 Why not just deploy the Helm chart directly from a `HelmRelease` stored in Git, a.k.a. the standard Flux workflow?

Because the sub-HelmReleases need their values templated, their existence 'if'd, they might be dynamic,...

Lots of reasons, basically the same reasons why you use helm in the first place and don't create `deployment.yaml`s by hand 😁
